compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -103,8 +103,13 @@ impl<'tcx> Context for Tables<'tcx> {
     }
 
     fn ty_kind(&mut self, ty: crate::stable_mir::ty::Ty) -> TyKind {
-        let ty = self.types[ty.0];
-        ty.stable(self)
+        self.types[ty.0].clone().stable(self)
```

Comment:
why do we need this clone now?

Reply:
It's just a `Ty<'tcx>`, so we should probably copy it.

The reason we need it is that `stable` takes `self` by mutable reference, while we immutably borrow from `self` via the index operations

Reply:
Actually I got this wrong. We need to clone the `TyKind` in case it comes out of a `MaybeStable::Stable` entry of `self.types`. Sometimes SMIR creates its own types, and they get interned that way.